core/src/main/java/tc/oc/pgm/modules/StatsMatchModule.java

```diff
@@ -0,0 +1,233 @@
+package tc.oc.pgm.modules;
+
+import java.text.DecimalFormat;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import net.md_5.bungee.api.ChatColor;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchModule;
+import tc.oc.pgm.api.match.MatchScope;
+import tc.oc.pgm.api.match.event.MatchFinishEvent;
+import tc.oc.pgm.api.match.factory.MatchModuleFactory;
+import tc.oc.pgm.api.module.exception.ModuleLoadException;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.api.player.PlayerRelation;
+import tc.oc.pgm.api.player.event.MatchPlayerDeathEvent;
+import tc.oc.pgm.events.ListenerScope;
+import tc.oc.pgm.tracker.damage.ProjectileInfo;
+import tc.oc.util.bukkit.component.Component;
+import tc.oc.util.bukkit.component.ComponentUtils;
+import tc.oc.util.bukkit.component.Components;
+import tc.oc.util.bukkit.component.types.PersonalizedText;
+import tc.oc.util.bukkit.component.types.PersonalizedTranslatable;
+import tc.oc.util.bukkit.nms.NMSHacks;
+
+@ListenerScope(MatchScope.RUNNING)
+public class StatsMatchModule implements MatchModule, Listener {
+
+  public static class Factory implements MatchModuleFactory<StatsMatchModule> {
+
+    @Override
+    public StatsMatchModule createMatchModule(Match match) throws ModuleLoadException {
+      return new StatsMatchModule();
+    }
+  }
+
+  public static class playerStats {
+    private int kills = 0;
+    private int deaths = 0;
+    private int killstreak = 0;
+    private int killstreakMax = 0;
+    private int longestBowKill = 0;
+
+    public void onMurder() {
+      kills++;
+      killstreak++;
+      if (killstreak > killstreakMax) killstreakMax = killstreak;
+    }
+
+    public void onDeath() {
+      deaths++;
+      killstreak = 0;
+    }
+
+    public void setLongestBowKill(double distance) {
+      if (distance > longestBowKill) {
+        longestBowKill = Integer.parseInt(new DecimalFormat("#").format(Math.round(distance)));
+      }
+    }
+
+    public Component getBasicStatsMessage() {
+      String KD;
+      if (deaths == 0) {
+        KD = "0";
+      } else {
+        DecimalFormat df = new DecimalFormat("#.##");
```

Comment:
Can we change it to a constant in this class? Also, call it `decimalFormat` because `df` is very cryptic.

Reply:
when you have `df.format(number);` it being called df isn't very cryptic tbh

Reply:
> Can we change it to a constant in this class? Also, call it `decimalFormat` because `df` is very cryptic.

What do you mean change it to a constant, the K/D variable?

Reply:
Make `DecimalFormat` a constant so it is not created each time it is used. Look that this is going to parse your input string (`#.##`) each time to create a new `DecimalFormat` instance.

Reply:
Yes of course!